File: Kotlin/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.kt
class Solution {
    var jokaBabb: Array<String> = arrayOf("aya", "ye", "woo", "ma")
    
    fun canSpeak(word: String): Boolean {
        for (j in jokaBabb) {
            if (word.length == 0){
                return true
            }
            if (word.startsWith(j)) {
                return canSpeak(word.substring(j.length))
            }
        }
        return false
    }
    
    fun solution(babbling: Array<String>): Int {
        var answer: Int = 0
        for (b in babbling) {
            if (canSpeak(b)) {
                answer += 1
            }
        }
        return answer
    }
}

File: Go/프로그래머스/0/120956. 옹알이 （1）/옹알이 （1）.go
import "strings"


var jokaBabb = []string{"aya", "ye", "woo", "ma"}

func checkCanSpeak(word string) bool {
    if len(word) == 0 {
        return true
    }
    for _, j := range jokaBabb {
        if strings.HasPrefix(word, j) {
            remainWord := word[len(j):]
            if checkCanSpeak(remainWord) {
                return true
            }
        }
    }
    return false
}

func solution(babbling []string) int {
    count := 0
    for _, b := range babbling {
        if checkCanSpeak(b) {
            count += 1
        }
    }
    return count
}